bundles/org.openhab.binding.doorbird/src/main/java/org/openhab/binding/doorbird/internal/DoorbirdHandler.java

```diff
@@ -0,0 +1,633 @@
+        updateDoorbellMontage();
+    }
+
+    // Callback used by listener to update motion channel
+    public synchronized void updateMotionChannel(long timestamp) {
+        logger.debug("Handler: Update MOTION channels for thing {}", getThing().getUID());
+        DoorbirdImage dbImage = downloadImage(buildUrl("/bha-api/image.cgi"));
+        if (dbImage != null) {
+            RawType image = dbImage.getImage();
+            updateState(CHANNEL_MOTION_IMAGE, image != null ? image : UnDefType.UNDEF);
+            updateState(CHANNEL_MOTION_TIMESTAMP, getLocalDateTimeType(dbImage.getTimestamp()));
+        }
+        triggerChannel(CHANNEL_MOTION, EVENT_TRIGGERED);
+        startMotionOffJob();
+        updateMotionMontage();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Got command {} for channel {} of thing {}", command, channelUID, getThing().getUID());
+
+        switch (channelUID.getId()) {
+            case CHANNEL_DOORBELL_IMAGE:
+                if (command instanceof RefreshType) {
+                    refreshDoorbellImageFromHistory();
+                }
+                break;
+            case CHANNEL_MOTION_IMAGE:
+                if (command instanceof RefreshType) {
+                    refreshMotionImageFromHistory();
+                }
+                break;
+            case CHANNEL_LIGHT:
+                handleLight(command);
+                break;
+            case CHANNEL_OPENDOOR1:
+                handleOpenDoor(command, "1");
+                break;
+            case CHANNEL_OPENDOOR2:
+                handleOpenDoor(command, "2");
+                break;
+            case CHANNEL_RESTART:
+                handleRestart(command);
+                break;
+            case CHANNEL_GETIMAGE:
+                handleGetImage(command);
+                break;
+            case CHANNEL_DOORBELL_HISTORY_INDEX:
+            case CHANNEL_MOTION_HISTORY_INDEX:
+                if (!(command instanceof RefreshType)) {
+                    handleHistoryImage(channelUID, command);
+                }
+                break;
+            case CHANNEL_GET_DOORBELL_MONTAGE:
+                if (command instanceof RefreshType || (command instanceof OnOffType && command.equals(OnOffType.ON))) {
+                    updateDoorbellMontage();
+                }
+                break;
+            case CHANNEL_GET_MOTION_MONTAGE:
+                if (command instanceof RefreshType || (command instanceof OnOffType && command.equals(OnOffType.ON))) {
+                    updateMotionMontage();
+                }
+                break;
+            case CHANNEL_SIP_HANGUP:
+                handleSipHangup(command);
+                break;
+        }
+    }
+
+    private void refreshDoorbellImageFromHistory() {
+        logger.debug("Handler: REFRESH doorbell image channel using most recent doorbell history image");
+        scheduler.schedule(() -> {
```

Comment:
```suggestion
        scheduler.execute(() -> {
```

also similar calls

Reply:
I'll fix all these. This was left over from when I had an initial delay on these. I had since remove the delay but failed to change to `execute`.